### PR TITLE
[WIP] Add initial API chapter

### PR DESCRIPTION
**Pull request by jezcope**
This is as far as I got during the Tuesday 4 June writing sprint, comments welcome!

### CODE REVIEW

File: book/_quarto.yml
Comment (noramcgregor) [inline book/_quarto.yml:1]: Don't worry about editing the quarto.yml, can you remove this from your pull request? I'll re-order things later. For now we'll leave it where it is in the hierarchy already....thanks!

File: book/api.qmd
Comment (noramcgregor) [inline book/api.qmd:1]: Please keep the header as is, this is a quarto book thing. So just revert the changes you made deleting lines 1-5 as the # What is an API? {.unnumbered} all has to stay etc....thanks!

Comment (jezcope) [reply book/api.qmd:1]: Sure, no problem!
